### ISSUE TITLE
How to use the trb file in combination with ProteinMPNN

### ISSUE THREAD

**Issue by wangdoc2002** [RosettaCommons/RFdiffusion#137]
Hi, RF diffusion team,
I am inpainting a protein with varied AA length.
For a fixed length of design, I can use --fixed_positions_jsonl function in ProteinMPNN. 

For a varied length of designs, the .trb file outputs which residues were fixed during RFdiffusion nicely.
Is there a way I can fed this trb information to ProteinMPNN?

**Comment by joewatchwell** [comment]
Yes! So the trb file contains the mappings of where residues were in the input, and where they ended up in the output.
These are "con_ref_idx0" and "con_hal_idx0" (named such for historical reasons).
Everything in "con_ref_idx0" are residues that were fixed in the input contig. "con_hal_idx0" contains the corresponding 0-indexed position of those residues in the output. This can then quite easily be used to then make the fixed_position dictionary inputs to ProteinMPNN.
We have scripts for doing this - if you have any difficulties we can probably share these

**Comment by yshen25** [comment]
It is a dictionary if you load the .trb file using pickle
